src/core/exceptions.h

```diff
@@ -8,12 +8,23 @@
  * This includes all public functions. This macro is meant to handle exceptions thrown
  * by C++ libraries used by igraph (such as bliss). Wrap the entire body
  * of public functions implemented in C++ in IGRAPH_HANDLE_EXCEPTIONS().
+ *
+ * In some cases IGRAPH_HANDLE_EXCEPTIONS() won't work because the
+ * C preprocessor gets confused by the code block. In that case one can use
+ * IGRAPH_HANDLE_EXCEPTIONS_BEGIN; and IGRAPH_HANDLE_EXCEPTIONS_END at the
+ * beginning and end of the code block.
  */
-#define IGRAPH_HANDLE_EXCEPTIONS(code) \
-    try { code; } \
+#define IGRAPH_HANDLE_EXCEPTIONS_BEGIN \
+    try {
+#define IGRAPH_HANDLE_EXCEPTIONS_END \
+    } \
     catch (const std::bad_alloc &e) { IGRAPH_ERROR(e.what(), IGRAPH_ENOMEM); /* LCOV_EXCL_LINE */ } \
     catch (const std::exception &e) { IGRAPH_ERROR(e.what(), IGRAPH_FAILURE); } \
     catch (...) { IGRAPH_ERROR("Unknown exception caught.", IGRAPH_FAILURE); }
 
+#define IGRAPH_HANDLE_EXCEPTIONS(code) \
+    IGRAPH_HANDLE_EXCEPTIONS_BEGIN; \
```

Comment:
This `;` is a bit strange here as it expands to `try {;`.

Reply:
feel free to remove the `;`, it has no effect. I just wanted to test that one is allowed to use `IGRAPH_HANDLE_EXCEPTIONS_BEGIN` or `_END` with or without a semicolon at the end without breaking thigs.